### PR TITLE
FIX: rewritten lib.tools_qgis.gqis_get_layer_source due to parsing er…

### PR DESCRIPTION
**Pull request by krj29b**
…rors in previous version.

On QGIS 3.4 worked fine, but we noticed that in QGIS 3.12+ layer.dataProvider().dataSourceUri() returns slightly different URI strings and causes parsing errors.

### CODE REVIEW

**Comment by krj29b** [comment]
this should solve issue  #771

File: lib/tools_qgis.py [hunk 1]
```diff
@@ -100,54 +101,28 @@ def qgis_get_layer_source(layer):
     """ Get database connection paramaters of @layer """
 
     # Initialize variables
-    layer_source = {'db': None, 'schema': None, 'table': None, 'service': None,
+    layer_source = {'db': None, 'schema': None, 'table': None,
```
Comment (daviderill) [inline lib/tools_qgis.py:104]: Key 'service' cannot be removed because is used in other file. Can you please add it again?

**Review by daviderill — CHANGES_REQUESTED** [review]
I've tested and it works perfectly. Thanks a lot! 
Just change this please:
Key 'service' cannot be removed because is used in other file. Can you please add it again?

**Review by daviderill — APPROVED** [review]

File: lib/tools_qgis.py (hunk 1 above)
Comment (krj29b) [reply lib/tools_qgis.py:104]: ok, i've corrected this in commit: 9947041c0ae0a84...

**Pull request merged**